stdlib/string.gr

```diff
@@ -1867,30 +1867,49 @@ export let forEachCodePointi = (fn: (Number, Number) -> Void, str: String) => {
   void
 }
 
+@unsafe
 let trimString = (str: String, end: Bool) => {
-  let chars = explode(str), charsLength = length(str)
-  let mut i = 0, offset = 1
+  let (+) = WasmI32.add
+  let (-) = WasmI32.sub
+  let (<) = WasmI32.ltU
+  let (!=) = WasmI32.ne
+
+  let mut stringPtr = WasmI32.fromGrain(str)
+  let charsLength = WasmI32.load(stringPtr, 4n)
+  stringPtr += 8n
+  let mut i = 0n, offset = 1n
   if (end) {
-    i = charsLength - 1
-    offset = -1
+    i = charsLength - 1n
+    offset = -1n
   }
-  for (; i < charsLength && i > -1; i += offset) {
-    let currentChar = chars[i]
+  let mut count = 0n
+  for (; i < charsLength; i += offset) {
+    // Get The Character
+    let byte = WasmI32.load8U(stringPtr, i)
     // TODO: Use unicode whitespace property and unicode line terminator once github issue #661 is completed
     if (
-      // Spacing
-      currentChar != '\u{0009}' && // Tab
-      currentChar != '\u{000B}' && // LINE TABULATION
-      currentChar != '\u{000C}' && // FORM FEED (FF)
-      currentChar != '\u{0020}' && // Space
-      currentChar != '\u{00A0}' && // No Break Space
-      currentChar != '\u{FEFF}' && // ZERO WIDTH NO-BREAK SPACE
+      byte != 0x09n && // Tab
+      byte != 0x0Bn && // LINE TABULATION
+      byte != 0x0Cn && // FORM FEED (FF)
+      byte != 0x20n && // Space
+      byte != 0xA0n && // No Break Space
+      byte != 0xFEFFn && // ZERO WIDTH NO-BREAK SPACE
```

Comment:
...which comes into play here, where this value is the code point value, but not what's loaded from the single byte. The UTF-8 representation is the byte sequence `EF BB BF`: https://unicode-table.com/en/FEFF/

You'll need to handle this case specially... If the value loaded is `0xEF` I'd probably just load a full word and check the bytes: `(WasmI32.load(stringPtr) >>> 8n) != 0xBFBBEF` (note that the bytes are in reverse order because wasm is little-endian).

Reply:
Also we should add a test.